apollo-runtime/src/main/java/com/apollographql/apollo/ApolloClient.java

```diff
@@ -231,6 +249,8 @@ Response cachedHttpResponse(String cacheKey) throws IOException {
     Optional<Logger> logger = Optional.absent();
     final List<ApolloInterceptor> applicationInterceptors = new ArrayList<>();
     boolean sendOperationIdentifiers;
+    String subscriptionServerUrl;
+    WebSocket.Factory webSocketFactory;
```

Comment:
How do you feel about making ApolloClient to have `SubscriptionTransport.Factory` directly to abstract WebSocket (implementation detail) from subscription API?

Reply:
Sorry I didn't get it, you want ApolloClient to implement `SubscriptionTransport.Factory`?

Reply:
Ah, sorry no -- I meant ApolloClient might hold reference to SubscriptionTransport.Factory directly rather than explicit reference to WebSocket.Factory